source/Background/Controller.js

```diff
@@ -379,15 +341,18 @@ backgroundController.exposeController(
 
       storage.set({ apps: newApps });
     });
-
-    if (response) {
+    if (response?.status === CONNECTION_STATUS.accepted) {
       try {
         const publicKey = await keyring.getPublicKey();
         callback(null, publicKey, [{ portId, callId }]);
+        callback(null, true);
       } catch (e) {
+        callback(ERRORS.SERVER_ERROR(e), null);
         callback(ERRORS.SERVER_ERROR(e), null, [{ portId, callId }]);
       }
     } else {
+      plugProvider.deleteAgent();
+      callback(ERRORS.AGENT_REJECTED, null);
       callback(ERRORS.AGENT_REJECTED, null, [{ portId, callId }]);
     }
```

Comment:
Seems that the clause is only necessary for the exception. So suggested:

```suggestion
    if (response?.status !== CONNECTION_STATUS.accepted) {
          plugProvider.deleteAgent();
          callback(ERRORS.AGENT_REJECTED, null);
          callback(ERRORS.AGENT_REJECTED, null, [{ portId, callId }]);
          return;
    }
    
    try {
      const publicKey = await keyring.getPublicKey();
      callback(null, publicKey, [{ portId, callId }]);
      callback(null, true);
    } catch (e) {
      callback(ERRORS.SERVER_ERROR(e), null);
      callback(ERRORS.SERVER_ERROR(e), null, [{ portId, callId }]);
    }
```